### ISSUE TITLE
Understanding the pytorch random seed assignment across ranks

### ISSUE THREAD

**Issue by Qianli-ion** [NVlabs/stylegan2-ada-pytorch#127]
First thanks for the great work! I have a quick question regarding the random seed assignment across GPUs (ranks). 

According to this line
https://github.com/NVlabs/stylegan2-ada-pytorch/blob/d4b2afe9c27e3c305b721bc886d2cb5229458eba/training/training_loop.py#L126
my understanding is that the torch random seed is manually set to be different across GPUs (with the + rank). This is a bit different from my prior experience which set the random seed to be the same across GPUs. Wouldn't different seeds lead to different weight initialization across GPUs? If so, can I ask the underlying intuition behind this implementation? 

Thanks for the help!

**Comment by shahviraj** [comment]
When DDP is used, pytorch internally takes care of initializing all the model replicas on different GPUs with the exact same weights. From pytorch documentation: 

"Construction: The DDP constructor takes a reference to the local module, and broadcasts state_dict() from the process with rank 0 to all other processes in the group to make sure that all model replicas start from the exact same state. "

https://pytorch.org/docs/master/notes/ddp.html

so the seed set as above would affect the other randomly initialized parameters (such as data sampling), but won't affect the weight initialization.
